### PR TITLE
[Merged by Bors] - docs: improve `Set` documentation

### PR DESCRIPTION
**Pull request by alexkeizer**
Document the fact that the defeq between `Set α` and `α → Prop` is an internal implementation detail

---

Inspired by: https://leanprover.zulipchat.com/#narrow/stream/113489-new-members/topic/Predicate.20vs.20Set, 
and the fact that I did not know this defeq was not to be used until recently.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by alreadydone — APPROVED** [review]

**Comment by alexjbest** [comment]
LGTM

maintainer merge

File: Mathlib/Init/Set.lean
Comment (Vierkantor) [inline Mathlib/Init/Set.lean:42]: I believe indented text in docstrings will be parsed as a Markdown code block.
```suggestion
Although `Set` is defined as `α → Prop`, this is an implementation detail which should not be
relied on. Instead, `setOf` and membership of a set (`∈`) should be used to convert between sets
and predicates.
```

**Review by Vierkantor — APPROVED** [review]
Thanks!

bors d+

**Comment by Vierkantor** [comment]
It looks like you didn't apply my suggestion before merging. Once doc-gen builds the new documentation, I'll check how it looks and open a fixing PR if needed. Next time, please check the review comments a bit more carefully.

**Comment by alexkeizer** [comment]
Sorry about that! I did not realize the app hides review comments by default, so I had not noticed your suggestion. I'll be more careful next time

**Pull request closed without merging**
